src/pages/ItemList.js

```diff
@@ -74,38 +77,70 @@ function ItemList({ token }) {
     fetchDocs(token);
   };
 
+  useEffect(() => {
+    const filterList = (e) => {
+      let lowerCase = searchInput.toLowerCase();
+      console.log(lowerCase);
+
+      const filteredData = docs.filter((item) => {
+        //if no input the return the original
+        if (searchInput === '') {
+          return item;
+        }
+        //return the item which contains the user input
+        else {
+          return item.item_name.toLowerCase().includes(lowerCase);
+        }
+      });
+      return filteredData;
+    };
+    setFilteredList(filterList());
+  }, [searchInput, docs]);
```

Comment:
To go off what Mike was saying about added state - is it possible to setDocs(filterList()) in this useEffect or would it mess with the onClick? 

I love the simplicity of the if/else. I was thinking how I would do it and my initial ideas were way more complicated lol.

Reply:
Thanks! `setDocs` is only used to set `docs` which is the list that comes from the database. Currently, we're only pulling it once when the page loads (which is the only db call we need at the moment). This useEffect runs on every keypress and filters that array and updates the filtered array every time you hit a key so I don't think we can put it inside. We haven't made the changes yet, but I believe the useEffect will be going away and we'll just use filterList(docs).map where list.map currently lives.